docs/modules/ROOT/pages/rpc.adoc

```diff
@@ -93,148 +93,148 @@ pub const ROTATE_ON_ERROR_CODES: [u16; 1] = [429];
+2. The `create_retryable_http_client` utility takes this base client and an `HttpRetryConfig` to produce a `ClientWithMiddleware`.
+3. This `ClientWithMiddleware` (the "retryable client") is then used for all HTTP operations within `HttpTransportClient`, including initial health checks, requests sent via `EndpointManager`, and `try_connect` calls during rotation. This ensures all operations benefit from the configured retry policy and the shared connection pool.
 
 Each transport client may define its own retry policy:
 
 [source,rust]
 ----
-
 // src/services/transports/http.rs
 pub struct HttpTransportClient {
-  pub client: Arc<RwLock<Client>>,
+  pub client: ClientWithMiddleware,
   endpoint_manager: EndpointManager,
   test_connection_payload: Option<String>,
 }
 
 // Example of client creation with retry mechanism
-let http_client = reqwest::ClientBuilder::new()
+// Use default retry policy
+let http_retry_config = HttpRetryConfig::default();
+
+// Create the base HTTP client
+let base_http_client = reqwest::ClientBuilder::new()
   .pool_idle_timeout(Duration::from_secs(90))
   .pool_max_idle_per_host(32)
   .timeout(Duration::from_secs(30))
   .connect_timeout(Duration::from_secs(20))
-  .build()?;
-
-// Create middleware client with retry policy
-let client = ClientBuilder::new(cloned_http_client)
-  .with(RetryTransientMiddleware::new_with_policy_and_strategy(
-    retry_policy,
-    RetryTransient,
-  ))
-  .build();
-
-// src/services/transports/evm/http.rs
-pub struct EVMTransportClient {
-  http_client: HttpTransportClient,
-}
+  .build()
+  .context("Failed to create base HTTP client")?;
 
-// override with a custom retry policy and strategy
-pub async fn new(network: &Network) -> Result<Self, anyhow::Error> {
-  let test_connection_payload = Some(r#"{"id":1,"jsonrpc":"2.0","method":"net_version","params":[]}"#.to_string());
-  let http_client = HttpTransportClient::new(network, test_connection_payload).await?;
-  http_client.set_retry_policy(
-    ExponentialBackoff::builder().build_with_total_retry_duration(Duration::from_secs(10)),
-    Some(DefaultRetryableStrategy),
-  )?;
-  Ok(Self { http_client })
-}
-----
+// Create a retryable HTTP client with the base client and retry policy
+let retryable_client = create_retryable_http_client(
+  &http_retry_config,
+  base_http_client,
+  Some(TransientErrorRetryStrategy), // Use custom or default retry strategy
+);
 
 === Implementation Details
-This retry and rotation strategies ensure optimal handling of different types of failures while maintaining service availability.
+The `EndpointManager` uses the retry-enabled `ClientWithMiddleware` provided by `HttpTransportClient` for its attempts on the primary URL. If these attempts (including internal `reqwest-retry` retries) ultimately fail with an error that warrants rotation (e.g., a 429 status code, or persistent network errors), then `EndpointManager` initiates the URL rotation sequence.
 
 [mermaid,width=100%]
```

Comment:
Looks like this is not properly formatted, the mermaid diagram is mixed in with the code-block, can you have a look?

![Screenshot 2025-06-13 at 11 57 27](https://github.com/user-attachments/assets/3c841bca-cafc-4bcf-8ae0-107cae6210c1)